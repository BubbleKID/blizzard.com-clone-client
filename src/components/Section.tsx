interface Props {
    backgroundImage: string
}

export default function Section(props: Props) {
    return(
        <div style={{height: '336px', backgroundImage: `url(${props.backgroundImage})`}}>
            im section
        </div>
    )
}