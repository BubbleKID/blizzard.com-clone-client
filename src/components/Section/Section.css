.section-container {
    background-position: center;
}

.section-wrapper {
    display: flex;
    justify-content: space-between;
    max-width: 1600px;
    margin: 0 auto;
}

.section-content {
    min-height: 300px;
    display: flex;
    flex-direction: column;
    padding: 56px;
    text-align: left;
    justify-content: center;
    align-items: flex-start;
}

.section-btn {
    margin-top: 16px;
    background-color: #006fb2;
    border: solid 1px#00aeff;
    display: block;
    color: #fff;
    font-size: 18px;
    font-weight: 400;
    padding: 10px 50px;
    text-decoration: none;
    text-align: center;
    cursor: pointer;
    outline-offset: 2px;
    display: inline-block;
    margin-right: 16px;
}

.transparent-btn {
    color: #00aeff;
    background-color: transparent;
    padding-left: 20px;
    padding-right: 20px;
}

.section-icon {
    display: inline-block;
    margin-right: 12px;
    align-items: center;
    margin: 8px 0;
    margin-right: 12px;
    width: 21.6px;
    height: 21.6px;
}

.section-icon path{
   fill: #00aeff;
}

.section-icon-inside-btn {
    width: 32.4px;
    height: 32.4px;
    margin: 0;
    margin-left: 8px;
}

.section-icon-inside-btn path{
    fill: #fff;
}

.section-title {
    font-size: 1.5rem;
    font-weight: 400;
    margin: 8px 0;
    padding: 0;
    display: inline-block;
}

.section-title-container {
    display: inline-flex;
    align-items: center;
    padding-top: 4px;
    padding-bottom: 4px;
}

.mobile-link {
    line-height: 1.5;
    color: #00aeff;
    text-decoration: underline;
    transition: color 200ms;
    text-underline-position: under;
}

.section-icon-large {
    transform: translateX(+50%);
    height: 312px;
    color: #148eff;
    opacity: .3;
}

.section-icon-large-container {
    display: flex;
    align-items: center;
}

.section-image-container{
    z-index: 1;
    display: flex;
    align-items: center;
}

.section-image-container img{
    height: 332px;
    width: auto;
}