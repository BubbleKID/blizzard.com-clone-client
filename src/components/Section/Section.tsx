import React from 'react';
import './Section.css';

interface Props {
    children: any
    height: string
    backgroundImages: string[]
}

export default function Section(props: Props) {
    return(
        <div className="section-container" style={{ height: props.height, backgroundImage: `url(${props.backgroundImages[0]})`}}>
            {props.children}
        </div>
    )
}