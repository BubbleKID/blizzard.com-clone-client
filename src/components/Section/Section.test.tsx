import React from 'react';
import TestRenderer  from 'react-test-renderer';
import Section from './Section';
import diablo4 from '../../images/game-card_diablo4_enUS.jpeg';

test('Card should be rendered', () => {
    const component = TestRenderer.create(<Section height="300px" backgroundImages={[diablo4, diablo4]}>children</Section>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});