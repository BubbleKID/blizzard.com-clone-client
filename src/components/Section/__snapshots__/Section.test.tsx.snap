// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should be rendered 1`] = `
<div
  className="section-container"
  style={
    Object {
      "backgroundImage": "url()",
      "height": "300px",
    }
  }
>
  children
</div>
`;
